pandas/core/arrays/masked.py

```diff
@@ -229,6 +231,30 @@ def to_numpy(
             data = self._data.astype(dtype, copy=copy)
         return data
 
+    def astype(self, dtype: Dtype, copy: bool = True) -> ArrayLike:
+        dtype = pandas_dtype(dtype)
+
+        if is_dtype_equal(dtype, self.dtype):
+            if copy:
+                return self.copy()
+            return self
+
+        # if we are astyping to another nullable masked dtype, we can fastpath
+        if isinstance(dtype, BaseMaskedDtype):
+            # TODO deal with NaNs for FloatingArray case
+            data = self._data.astype(dtype.numpy_dtype, copy=copy)
+            # mask is copied depending on whether the data was copied, and
+            # not directly depending on the `copy` keyword
+            mask = self._mask if data is self._data else self._mask.copy()
```

Comment:
I wrote this code myself in the past, but I am not actually sure now this is needed. AFAIK the `self._data.astype` (numpy's method) will always return a copy unless `dtype.numpy_dtype` is the exact same type as the data. But that case should already be covered by the `if is_dtype_equal(dtype, self.dtype)` above. 

(now since this is copying existing code, fine to leave this for another issue)

Reply:
makes sense, and i think you're right about the numpy behavior